### PR TITLE
Pass Build compatible extension TCK tests

### PR DESCRIPTION
**Pull request by graemerocher**
Passes all the build compatible extension tests exception the ones that check for errors

### CODE REVIEW

File: tck-runner/passingTests.xml [hunk 1]
```diff
@@ -1235,110 +1235,52 @@
             </class>
             <!--            Extensions-->
             <class name="org.jboss.cdi.tck.tests.build.compatible.extensions.customStereotype.CustomStereotypeTest">
```
Comment (dstepanov) [inline tck-runner/passingTests.xml]: If the tests are passing delete those entries

File: cdi/src/main/java/com/oracle/odi/cdi/OdiInstanceImpl.java [hunk 2]
```diff
@@ -55,46 +56,67 @@
     private Map<T, CreationalContext<T>> created = new HashMap<>();
 
     OdiInstanceImpl(BeanContext beanContext,
-                    OdiBeanContainer beanContainer,
+                    OdiBeanContainerImpl beanContainer,
```
Comment (dstepanov) [inline cdi/src/main/java/com/oracle/odi/cdi/OdiInstanceImpl.java]: Is there a way not to pass implementations?

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: cdi/src/main/java/com/oracle/odi/cdi/OdiBeanContainerImpl.java [hunk 3]
```diff
@@ -217,12 +225,96 @@ public <T> Set<ObserverMethod<? super T>> resolveObserverMethods(T event, Annota
...
         Argument<?> argument = Argument.of(event.getClass());
-        final io.micronaut.context.Qualifier qualifierInstances = AnnotationUtils
-                .qualifierFromQualifierAnnotations(qualifiers);
+        final io.micronaut.context.Qualifier qualifierInstances =
+                qualifierFromQualifierAnnotations(qualifiers);
         return observerMethodRegistry
                 .findSetOfObserverMethods(argument, qualifierInstances);
     }
 
+    <T1> Qualifier<T1> resolveQualifier(Annotation[] annotations) {
```
Comment (dstepanov) [inline cdi/src/main/java/com/oracle/odi/cdi/OdiBeanContainerImpl.java]: Can this and method below be moved to some utils class?

Comment (graemerocher) [reply cdi/src/main/java/com/oracle/odi/cdi/OdiBeanContainerImpl.java]: problem is the set of qualifiers is extensible via extensions so I have computed and populated them into a type called `RuntimeMetaAnnotations`. I don't know how to make this available statically

File: cdi/src/main/java/com/oracle/odi/cdi/OdiInstanceImpl.java (hunk 2 above)
Comment (graemerocher) [reply cdi/src/main/java/com/oracle/odi/cdi/OdiInstanceImpl.java]: Will investigate

File: cdi/src/main/java/com/oracle/odi/cdi/OdiBeanContainerImpl.java (hunk 3 above)
Comment (dstepanov) [reply cdi/src/main/java/com/oracle/odi/cdi/OdiBeanContainerImpl.java]: Maybe extract it in some class that can be
possibly exposed from the bean container interface if needed.

File: tck-runner/passingTests.xml (hunk 1 above)
Comment (dstepanov) [reply tck-runner/passingTests.xml]: Going to rename the xml file to prevent confusion

**Review by dstepanov — APPROVED** [review]

**Pull request merged**